CalibCalorimetry/CaloTPG/src/CaloTPGTranscoderULUT.h

```diff
@@ -62,13 +62,17 @@ class CaloTPGTranscoderULUT : public CaloTPGTranscoder {
 
   // Member functions
   unsigned int getOutputLUTSize(const HcalTrigTowerDetId& id) const;
+  bool isOnlyQIE11(const HcalTrigTowerDetId& id) const;
   void loadHCALCompress(HcalLutMetadata const&, HcalTrigTowerGeometry const&) ; //Analytical compression tables
 
   // Member Variables
+  bool allLinear_ = false;
   double nominal_gain_;
   double lsb_factor_;
   double rct_factor_;
   double nct_factor_;
+  double lin8_factor_;
+  double lin11_factor_;
```

Comment:
see my comment above concerning the configuration parameters

Reply:
**That** parameter is set by the `setup(…)` function though. How do I handle that better?

Reply:
Ok, looks like `setup` needs the EventSetup, i.e. you cannot know the value of these parameters at construction time.

Reply:
I just tried to remove the `setup` function in its entirety and move the logic to the constructor. There's another piece of code that depends on this class from L1 that in theory should be broken anyways? If we can remove that, it'll make the code a bit cleaner.